src/main/kotlin/no/nav/syfo/service/mikrofrontend/MikrofrontendAktivitetskravService.kt

```diff
@@ -0,0 +1,63 @@
+package no.nav.syfo.service.mikrofrontend
+
+import no.nav.syfo.db.DatabaseInterface
+import no.nav.syfo.db.domain.PMikrofrontendSynlighet
+import no.nav.syfo.db.fetchFnrsWithExpiredMicrofrontendEntries
+import no.nav.syfo.db.fetchMikrofrontendSynlighetEntriesByFnr
+import no.nav.syfo.db.updateMikrofrontendEntrySynligTomByFnrAndTjeneste
+import no.nav.syfo.kafka.consumers.varselbus.domain.ArbeidstakerHendelse
+import no.nav.syfo.kafka.consumers.varselbus.domain.HendelseType
+import no.nav.syfo.kafka.consumers.varselbus.domain.getSynligTom
+import no.nav.syfo.kafka.producers.minsideMikrofrontend.MinSideRecord
+import no.nav.syfo.kafka.producers.minsideMikrofrontend.Tjeneste
+import no.nav.syfo.service.mikrofrontend.MikrofrontendService.Companion.actionEnabled
+import no.nav.syfo.utils.DuplicateAktivitetskravException
+import org.slf4j.LoggerFactory
+
+class MikrofrontendAktivitetskravService(val database: DatabaseInterface) {
+    val mikrofrontendId = "syfo-aktivitetskrav"
+
+    fun createOrUpdateAktivitetskravMicrofrontendByHendelse(hendelse: ArbeidstakerHendelse): MinSideRecord? {
+        return createOrUpdateMinSideRecord(hendelse)
+    }
+
+    private fun createOrUpdateMinSideRecord(hendelse: ArbeidstakerHendelse): MinSideRecord? {
+        val isMikrofrontendActiveForUser =
+            existingMikrofrontendEntries(hendelse.arbeidstakerFnr, Tjeneste.AKTIVITETSKRAV).isNotEmpty()
+        if (!isMikrofrontendActiveForUser && hendelse.shouldActivateMikrofrontend()) {
+            return minSideRecordEnabled(hendelse.arbeidstakerFnr)
+        } else if (isMikrofrontendActiveForUser || hendelse.shouldActivateMikrofrontend()){
```

Comment:
Kunne man droppet den OR-en der, || hendelse.shouldActivateMikrofrontend() ?

Den vil vel kanskje få true på første biten alltid uansett?

Reply:
Og dersom første biten skulle være false (mikrofrontend er ikke enablet), så blir det kanskje rart å sette ny expiry date?

Reply:
@AudunSorheim Om `!isMikrofrontendActiveForUser && hendelse.shouldActivateMikrofrontend()` er `false` betyr ikke det at `hendelse.shouldActivateMikrofrontend()` er `true` (Hvis `isMikrofrontendActiveForUser` er `false` kan også `hendelse.shouldActivateMikrofrontend()` være `false`. Vi må ha OR'en for å sørge for at gjentatte FORHAANDSVARSEL-events fører til at MF forlenger visningstiden.

Reply:
Går litt saktere her nå for min del :D Prøver å gjøre det om til tekst:
1. Hvis mikrofrontend ikke er skrudd på, og man får forhåndsvarsel eller NY, så skru på.
2. Hvis mikrofrontend er skrudd på, eller man får forhåndsvarsel eller NY, forleng visningstid

Det som skurrer for min del på del 2 er at dersom mikrofrontend er skrudd av, og du får forhåndsvarsel eller ny, så vil vel dette inntreffe på punkt 1, og aldri komme til punkt 2? Blir ikke det slik at dersom du havner på else if, så setter vi ny tid uansett så lenge mikrofrontend er aktiv?

Reply:
Ikke noen big deal altså, bare lurte på om det var jeg som tenkte feil eller ikke :)

Reply:
@AudunSorheim Du har helt rett! Det er smør på flesk med OR. Måtte tenke en stund på det. Skal få sløyfet den bort 👍